frontend/src/routes/Home/Overview/overviewDataFunctions.tsx

```diff
@@ -145,65 +145,31 @@ export function getNodeSummary(filteredClusters: Cluster[], t: TFunction<string,
   }
 }
 
-function getApps(
-  applications: Application[],
-  applicationSets: ApplicationSet[],
-  argoApps: ArgoApplication[],
-  discoveredApps: ArgoApplication[],
-  ocpAppResources: any[],
-  filteredClusterNames: string[],
-  allClusters: Cluster[],
-  placementDecisions: PlacementDecision[],
-  subscriptions: Subscription[]
-) {
-  const apps = [...applications, ...applicationSets, ...argoApps, ...discoveredApps, ...ocpAppResources]
-  // If no cluster labels are selected we default to the filteredClusterNames containing all cluster names
-  if (allClusters.length > filteredClusterNames.length) {
-    // filter apps by clusters from label selection.
-    return apps.filter((app) => {
-      const localCluster = allClusters.find((cls) => cls.name === 'local-cluster')
-      const clusterList = getClusterList(app, argoApps, placementDecisions, subscriptions, localCluster, allClusters)
+export function getAppTypeSummary(requestedCounts: IResultStatuses, t: TFunction<string, undefined>) {
+  const { itemCount = 0, filterCounts = { type: {} }, loading } = requestedCounts || {}
 
-      return filteredClusterNames.some((value) => {
-        return clusterList.includes(value)
-      })
-    })
+  const getAppTypeLabel = (type: string) => {
+    switch (type) {
+      case 'openshift':
+        return 'OpenShift'
+      case 'flux':
+        return t('Flux')
+      case 'subscription':
+        return t('Subscription')
+      case 'argo':
+        return t('Argo CD')
+      case 'appset':
+        return t('Application set')
+      case 'openshift-default':
+        return t('System')
+      default:
+        return t('Application')
+    }
   }
-  return apps
-}
-
-export function getAppTypeSummary(
-  applications: Application[],
-  applicationSets: ApplicationSet[],
-  argoApps: ArgoApplication[],
-  discoveredArgoApps: ArgoApplication[],
-  ocpAppResources: any[],
-  filteredClusterNames: string[],
-  allClusters: Cluster[],
-  placementDecisions: PlacementDecision[],
-  subscriptions: Subscription[],
-  t: TFunction<string, undefined>
-) {
-  const appTypes = getApps(
-    applications,
-    applicationSets,
-    argoApps,
-    discoveredArgoApps,
-    ocpAppResources,
-    filteredClusterNames,
-    allClusters,
-    placementDecisions,
-    subscriptions
-  ).map((app) => getApplicationType(app, t))
   const typeTotals: Record<string, number> = {}
-  appTypes.forEach((app) => {
-    if (typeof typeTotals[app] === 'undefined') {
-      typeTotals[app] = 1
-    } else {
-      typeTotals[app] += 1
-    }
+  Object.keys(filterCounts.type).forEach((type) => {
```

Comment:
Hi @jeswanke 
I believe there could be a bug here, as we're now integrating the Flight Control into the Overview page, we noticed that when switching the Fleet view, an error occurs and it seems to point to this line. In the debugger, I can see that `filterCounts` is just an empty object, with no `type` prop which breaks it.

![overview-switch](https://github.com/user-attachments/assets/de835fde-11aa-4a94-be0a-3ca287041c30)

Reply:
Hey @celdrake , John fixed this issue here: https://github.com/stolostron/console/pull/3876.